src/etc/inc/interfaces.inc

```diff
@@ -1128,34 +1128,49 @@ function interface_gre_configure(&$gre, $grekey = "") {
 		$greif = pfSense_interface_create("gre");
 	}
 
+	$tunnel_type = '';
+	if (($gre['tunnel-local-addr'] !== '') || ($gre['tunnel-remote-addr'] !== '')) {
+		$tunnel_type = 'v4';
+	}
+	if (($gre['tunnel-local-addr6'] !== '') || ($gre['tunnel-remote-addr6'] !== '')) {
```

Comment:
These should use `!empty()` and `&&` rather than `!== ''` and ``||``. and this will not give you the result you want if the values are unset. For example `null !== ''` is `true`. If either one is null, the test would be true and it would set the wrong type.